File: library/src/main/java/com/codehospital/versioning/VersioningExtension.java
package com.codehospital.versioning;

import org.gradle.api.Project;

/**
 * Gradle extension for accessing versioning functionality in build scripts.
 * This makes Versioning accessible as `versioning` in build.gradle.kts.
 */
public class VersioningExtension {
    public static final String NAME = "versioning";

    private final Project project;

    public VersioningExtension(Project project) {
        this.project = project;
    }

    public int getVersionCode(String buildType) {
        return Versioning.getVersionCode(project, buildType);
    }

    public String getVersionName(String buildType) {
        return Versioning.getVersionName(project, buildType);
    }

    public String getDebugSuffix() {
        return Versioning.getDebugSuffix(project);
    }

    public int getVersionBuild() {
        return Versioning.getVersionBuild(project);
    }

    public void incrementVersionBuild() {
        Versioning.incrementVersionBuild(project);
    }

    public void incrementVersionPatch() {
        Versioning.incrementVersionPatch(project);
    }
}

File: library/src/main/kotlin/com/codehospital/versioning/VersioningExtension.kt
package com.codehospital.versioning

import org.gradle.api.Project
import org.gradle.api.provider.Property
import org.gradle.api.provider.Provider

/**
 * Gradle extension for accessing versioning functionality in build scripts.
 * This makes Versioning accessible as `versioning` in build.gradle.kts
 */
abstract class VersioningExtension(private val project: Project) {

    fun getVersionCode(buildType: String = "debug"): Int {
        return Versioning.getVersionCode(project, buildType)
    }

    fun getVersionName(buildType: String = "release"): String {
        return Versioning.getVersionName(project, buildType)
    }

    fun getDebugSuffix(): String {
        return Versioning.getDebugSuffix(project)
    }

    fun getVersionBuild(): Int {
        return Versioning.getVersionBuild(project)
    }

    fun incrementVersionBuild() {
        Versioning.incrementVersionBuild(project)
    }

    fun incrementVersionPatch() {
        Versioning.incrementVersionPatch(project)
    }

    companion object {
        const val NAME = "versioning"
    }
}
